lib/api/CDataSummarizationJsonSerializer.cc

```diff
@@ -27,60 +28,93 @@
 
 #include <memory>
 #include <sstream>
+#include <stdexcept>
 #include <string>
 #include <utility>
 
 namespace ml {
 namespace api {
 
 namespace {
+using TBoolVec = std::vector<bool>;
+using TDoubleVec = std::vector<double>;
+using TStrVec = std::vector<std::string>;
+using TStrVecVec = std::vector<TStrVec>;
 using TRowItr = core::CDataFrame::TRowItr;
 using TFilteredInput = boost::iostreams::filtering_stream<boost::iostreams::input>;
 using Device = boost::iostreams::basic_array_source<char>;
 using TStreamBuffer = boost::iostreams::stream_buffer<Device>;
-using TStrVec = std::vector<std::string>;
 
-using TUint64Optional = boost::optional<std::uint64_t>;
+template<typename GET, typename VALUE>
+auto ifExists(const std::string& tag, const GET& get, const VALUE& value)
+    -> decltype(get(value[tag])) {
+    if (value.HasMember(tag)) {
+        try {
+            return get(value[tag]);
+        } catch (const std::runtime_error& e) {
+            LOG_ERROR(<< "Field '" << tag << "' " << e.what() << ".");
+        }
+    } else {
+        LOG_ERROR(<< "Field '" << tag << "' is missing.");
+    }
+    throw std::runtime_error{""};
```

Comment:
I think this should contain a unique message.  Even if it is never expected to be seen, if it leaks out one day because the logic is changed elsewhere then it will be nice to be able to find out where it came from.